docs/swift-3-api.diff

```diff
@@ -0,0 +1,399 @@
+--- /Users/austinzheng/Desktop/APIDUMP-2.2	2016-09-08 10:46:03.000000000 -0700
++++ /Users/austinzheng/Desktop/APIDUMP-3.0	2016-09-08 11:43:15.000000000 -0700
+@@ -1,117 +1,129 @@
+ public typealias NotificationToken = RLMNotificationToken
+ public typealias PropertyType = RLMPropertyType
+-public enum Error: ErrorType
+-    public var _code: Int
+-    public var _domain: String
+-public func == (lhs: ErrorType, rhs: ErrorType) -> Bool
+-    public func == (lhs: ErrorType, rhs: ErrorType) -> Bool
+-public func ~= (lhs: Error, rhs: ErrorType) -> Bool
+-    public func ~= (lhs: Error, rhs: ErrorType) -> Bool
++public struct Error
++    public enum Code : Int
++    public static let fail: Code = .fail
++    public static let fileAccess: Code = .fileAccess
++    public static let filePermissionDenied: Code = .filePermissionDenied
++    public static let fileExists: Code = .fileExists
++    public static let fileNotFound: Code = .fileNotFound
++    public static let incompatibleLockFile: Code = .incompatibleLockFile
++    public static let fileFormatUpgradeRequired: Code = .fileFormatUpgradeRequired
++    public static let addressSpaceExhausted: Code = .addressSpaceExhausted
++    public static let schemaMismatch: Code = .schemaMismatch
++    public var code: Code
++    public var _nsError: NSError
++    public init(_nsError error: NSError)
++public typealias _ErrorType = RLMError
++    public static var _nsErrorDomain = RLMErrorDomain
+ public class LinkingObjectsBase: NSObject, NSFastEnumeration
+-    public func countByEnumeratingWithState(state: UnsafeMutablePointer<NSFastEnumerationState>, objects buffer: AutoreleasingUnsafeMutablePointer<AnyObject?>, count len: Int) -> Int
++    public func countByEnumerating(with state: UnsafeMutablePointer<NSFastEnumerationState>, objects buffer: AutoreleasingUnsafeMutablePointer<AnyObject?>!, count len: Int) -> Int
+ public final class LinkingObjects<T: Object>: LinkingObjectsBase
+     public subscript(index: Int) -> T
+     public typealias Element = T
+     public var realm: Realm? 
+-    public var invalidated: Bool 
++    public var isInvalidated: Bool
+     public var count: Int 
+     public init(fromType type: T.Type, property propertyName: String)
+     public override var description: String
+-    public func indexOf(object: T) -> Int?
+-    public func indexOf(predicate: NSPredicate) -> Int?
+-    public func indexOf(predicateFormat: String, _ args: AnyObject...) -> Int?
++    public func index(of object: T) -> Int?
++    public func indexOfObject(for predicate: NSPredicate) -> Int?
```

Comment:
`for` feels odd here. It reads as "index of object for predicate". Something like "index of object matching predicate" would be clearer.